be/src/agent/task_worker_pool.cpp

```diff
@@ -197,6 +197,52 @@ void TaskWorkerPool::submit_task(const TAgentTaskRequest& task) {
     }
 }
 
+void TaskWorkerPool::submit_tasks(std::vector<TAgentTaskRequest>* tasks) {
+    DCHECK(!tasks->empty());
+    std::string type_str;
+    const TTaskType::type task_type = (*tasks)[0].task_type;
+    EnumToString(TTaskType, task_type, type_str);
+    {
+        std::lock_guard task_signatures_lock(_s_task_signatures_lock);
+        for (auto it = tasks->begin(); it != tasks->end();) {
+            TAgentTaskRequest task_req = *it;
+            int64_t signature = task_req.signature;
+            LOG(INFO) << "submitting task. type=" << type_str << ", signature=" << signature;
+
+            // batch register task info
+            std::set<int64_t>& signature_set = _s_task_signatures[task_type];
+            if (signature_set.insert(signature).second) {
+                (const_cast<TAgentTaskRequest&>(task_req)).__set_recv_time(time(nullptr));
```

Comment:
Why not to set the time in task_req is created?

Reply:
Because the meaning of recv_time should be when it is received, and task_req is created in agent_server, not the receiving time of task_worker_pool. However, it can be optimized in batches here.